File: Walkamib/app/src/main/java/com/example/walk_a_mib/repository/user/UserResponseCallback.java
package com.example.walk_a_mib.repository.user;

//import com.example.walk_a_mib.model.News;
import com.example.walk_a_mib.model.user.User;

public interface UserResponseCallback {
    void onSuccessFromAuthentication(User user);
    void onFailureFromAuthentication(String message);
    void onSuccessFromRemoteDatabase(User user);
    void onFailureFromRemoteDatabase(String message);
    void onSuccessLogout();
    void onSuccessFromPasswordReset();
    void onFailureFromPasswordReset(String message);
}

File: Walkamib/app/src/main/java/com/example/walk_a_mib/repository/user/UserResponseCallback.kt
package com.example.walk_a_mib.repository.user

import com.example.walk_a_mib.model.user.User

interface UserResponseCallback {
    fun onSuccessFromAuthentication(user: User?)
    fun onFailureFromAuthentication(message: String?)
    fun onSuccessFromRemoteDatabase(user: User?)
    fun onFailureFromRemoteDatabase(message: String?)
    fun onSuccessLogout()
    fun onSuccessFromPasswordReset()
    fun onFailureFromPasswordReset(message: String?)
}